smartsim/_core/entrypoints/redis.py

```diff
@@ -54,31 +58,64 @@ def handle_signal(signo: int, _frame: t.Optional[FrameType]) -> None:
     cleanup()
 
 
-def build_bind_args(ip_addresses: t.List[str]) -> t.List[str]:
-    bind_arg = f"--bind {' '.join(ip_addresses)}"
-    # pin source address to avoid random selection by Redis
-    bind_src_arg = f"--bind-source-addr {ip_addresses[0]}"
-    return [bind_arg, bind_src_arg]
+def build_bind_args(source_addr: str, *addrs: str) -> t.Tuple[str, ...]:
+    return (
+        "--bind",
+        source_addr,
+        *addrs,
+        # pin source address to avoid random selection by Redis
+        "--bind-source-addr",
+        source_addr,
```

Comment:
in the prior version, correctly or not, it built --bind-source-addr with the first address only and --bind with all addresses.

i like the idea of specifying the source_addr instead of assuming 0, but wonder if we need to ensure that source_addr is also in addrs (e.g. call was `x.build_bind_args(addrs[0], addrs[1:])`